api/instancemutater/machine.go

```diff
@@ -141,16 +166,15 @@ func (m *Machine) WatchApplicationLXDProfiles() (watcher.NotifyWatcher, error) {
 }
 
 type ProfileInfo struct {
-	Changes         bool
+	ModelName       string
+	InstanceId      instance.Id
 	ProfileChanges  []ProfileChanges
 	CurrentProfiles []string
 }
 
 type ProfileChanges struct {
```

Comment:
Suggest renaming to something like UnitProfileInfo{}. We also need the application name and the charm revision data here.

There's an lxdprofile.Profile{} we can use now instead of the local CharmLXDProfile{}. If so, can CharmLXDProfiles{} be removed?